### ISSUE TITLE
Should home screen widget form shortcut should link to latest version or pinned version?

### ISSUE THREAD

**Issue by florianm** [getodk/collect#4247]
## Software and hardware versions 
Collect v1.29.0 beta

#### Problem description
To help our enumerators remember which form to fill in their survey workflow, we use the ODK 1x1 widget to create shortcuts on the devices' home screen in a particular order.

Collect allows to remotely maintain devices through settings "exactly match server" and "hide old form versions". 
Currently, the widgets link to the latest form versions at the time of creation, and still do so even if the form has been replaced with newer versions.

This requires us to manually update each widget on each device whenever forms get updated.

#### Steps to reproduce the problem
* Publish a form through ODK Central. The form has a title, an ID, and an optional version.
* Create an ODK 1x1 widget on a device's home screen. The widget is named after the form title.
* Set ODK Collect to "exactly match server" and "hide old form versions".
* Publish an updated form version on  ODK Central. The updated form must have the same ID, an updated version string, and possibly an updated title.
* The widget remains named after the original form title and still opens the old form version.

#### Expected behavior
There's probably a good reason behind pinning form versions rather than form IDs in the ODK 1x1 widget.
However, for the use case "fully remote form updates", the widgets should link to the latest respective version (eg. using form ID) of each form instead of a pinned form version.

**Comment by lognaturel** [comment]
I haven’t confirmed yet but I’m guessing that the shortcut is based on a database ID that represents a specific formid/version combination. We may not have control over that. 

Is this as necessary with forms exactly matched from the server? Would a shortcut to the form list screen help? Can you name your forms with a numeric prefix to identify order? (E.g “1 - intake”, “2 - first encounter”, etc) The form list screen can be sorted by name. 

We do want to eventually make form order server, configurable but that is not on the immediate roadmap.

**Comment by florianm** [comment]
Based on the error messages I get when opening a shortcut to a freshly removed form, it's indeed some kind of integer ID linking a specific form.

We lay out our shortcuts on the home screen in a flowchart like pattern to bridge the mental gap between "what did I encounter now" and "what form to fill". We have a survey start and a survey end form that we show as first and last form on the screen (otherwise enumerators will inevitably forget to fill them out). Our forms are filled in an ever changing order (we have one for each entity type we encounter), and different campaigns use different set of forms, so numbering forms (and hiding some versions) is the analogue of releasing three fully greased up pigs numbered 1, 2, and 4 in a busy marketplace.

Therefore, the widgets are a wonderful way for us to arrange forms by non-linear workflow, while allowing logical and short form names.

**Comment by florianm** [comment]
Link to the [forum discussion](https://forum.getodk.org/t/odk-collect-v2021-2-beta-multiple-project-support-and-project-configuration-on-first-launch/33996/26)

We've tested and clarified our requirements:
We're updating forms through the managed QR codes. This is very useful, as our enumerators are spread across the state, and managing forms for our fleet of devices is a breeze.

Fixed:
* Form shortcuts are now stable enough to survive enabling/disabling other forms in the same project.

Remaining issues:

* The shortcuts does link to a fixed form version, but it should link to the latest version of a form.
* The 1x1 ODK widget menu does show all available form versions, but only shows the (same) [form name](https://github.com/seadowg/collect/blob/0fb11ff2484bcbafaaad07b3af48482261fd35ab/collect_app/src/main/java/org/odk/collect/android/activities/AndroidShortcutsActivity.java#L92). It should either hide old form versions (if selected in settings), and/or include form version and date updated, like the Fill Blank Form menu does.
